### ISSUE TITLE
OptionSetter.hpp: BoostWriteFunction is used as a type, but not defined in Types.hpp

### ISSUE THREAD

**Issue by adney11** [jpbarrette/curlpp#84]
#ifdef HAVE_BOOST
	/**
	* Specialization.
	*/
	template<>
	class OptionSetter<curlpp::types::**BoostWriteFunction**,											CURLOPT_HEADERFUNCTION>
	{
	public:
		typedef curlpp::types::BoostWriteFunction
			OptionValueType;
		typedef curlpp::internal::OptionContainerType<OptionValueType>::HandleOptionType
			ParamType;
		static void setOpt(internal::CurlHandle * handle, ParamType value);
	};
#endif // #ifdef HAVE_BOOST

Basically, Boost functors are not defined in Types.hpp, but are being used in OptionSetter.cpp
What is the fix for this?
